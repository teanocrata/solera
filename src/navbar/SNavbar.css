.Nav-logo {
  height: 100%;
}

.navbar-header a {
  height: 100%;
}

.navbar li{
  text-transform: uppercase;
  min-height: 110px;
  display: flex;
  align-items: center;
}

.navbar-inverse {
  background-image: none;
  background-color: lightgreen;
  border-color: transparent;
}

.navbar-inverse .navbar-nav > li > a {
    color: forestgreen;
}

.navbar-inverse .navbar-brand, .navbar-inverse .navbar-nav > li > a {
    text-shadow: none;
}
