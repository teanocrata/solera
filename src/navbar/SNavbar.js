import React, { Component } from "react";

import { Navbar, Nav, NavItem, NavDropdown, MenuItem } from "react-bootstrap";
import "./SNavbar.css";

class SNavbar extends Component {
  render() {
    return (
      <Navbar inverse collapseOnSelect fixedTop>
        <Navbar.Header>
          <Navbar.Brand>
            <a href="/">
              <img className="Nav-logo" src={this.props.logo} alt="Logo" />
            </a>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          {this.props.items ? (
            <Nav pullRight>
              {this.props.items.map(
                item =>
                  item.link ? (
                    <NavItem key={item.text} eventKey={1} href={item.link}>
                      {item.text}
                    </NavItem>
                  ) : (
                    <NavDropdown
                      eventKey={3}
                      key={item.text}
                      title={item.text}
                      id="basic-nav-dropdown"
                    >
                      {item.items.map(innerItem => (
                        <MenuItem
                          key={innerItem.text}
                          eventKey={1}
                          href={innerItem.link}
                        >
                          {innerItem.text}
                        </MenuItem>
                      ))}
                    </NavDropdown>
                  )
              )}
            </Nav>
          ) : (
            ""
          )}
        </Navbar.Collapse>
      </Navbar>
    );
  }
}

export default SNavbar;
