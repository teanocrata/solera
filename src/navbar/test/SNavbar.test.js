import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SNavbar from "../SNavbar";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SNavbar />, div);
});

test("SNavbar renders correctly", () => {
  const app = renderer.create(<SNavbar />).toJSON();
  expect(app).toMatchSnapshot();
});
