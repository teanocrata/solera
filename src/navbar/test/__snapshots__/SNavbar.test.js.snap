// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNavbar renders correctly 1`] = `
<nav
  className="navbar navbar-inverse navbar-fixed-top"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt="Logo"
          className="Nav-logo"
          src="solera_logo_80x80_green.png"
        />
      </a>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <ul
        className="nav navbar-nav navbar-right"
        role={null}
      >
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="https://www.clasesdeequitacionmadrid.com/instalaciones-hipica"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Instalaciones
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="https://www.clasesdeequitacionmadrid.com/clases-de-equitacion"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Clases
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="https://www.clasesdeequitacionmadrid.com/pupilaje"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Pupilaje
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href="#"
            id="basic-nav-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Actividades
             
            <span
              className="caret"
            />
          </a>
          <ul
            aria-labelledby="basic-nav-dropdown"
            className="dropdown-menu"
            role="menu"
          >
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
                xlinkHref="https://www.clasesdeequitacionmadrid.com/cumpleanos-con-caballos"
              >
                Cumplea√±os
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
                xlinkHref="https://www.clasesdeequitacionmadrid.com/cumpleanos-con-caballos"
              >
                Campamentos
              </a>
            </li>
          </ul>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="https://www.clasesdeequitacionmadrid.com/blog"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Blog
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
